components/layout_2020/sizing.rs

```diff
@@ -107,17 +107,37 @@ impl BoxContentSizes {
         &self,
         style: &ComputedValues,
     ) -> (ContentSizes, Percentage) {
-        // FIXME: account for 'min-width', 'max-width', 'box-sizing'
-
+        // FIXME: account for 'box-sizing'
         let inline_size = style.box_size().inline;
+        let min_inline_size = style
+            .min_box_size()
+            .inline
+            .percentage_relative_to(Length::zero())
+            .auto_is(Length::zero);
+        let max_inline_size = style
+            .max_box_size()
+            .inline
+            .to_option()
+            .and_then(|lp| lp.as_length());
+        let clamp = |l: Length| l.clamp_between_extremums(min_inline_size, max_inline_size);
+
         // Percentages for 'width' are treated as 'auto'
         let inline_size = inline_size.map(|lp| lp.as_length());
         // The (inner) min/max-content are only used for 'auto'
         let mut outer = match inline_size.non_auto().flatten() {
```

Comment:
Can't you use `and_then` to define `inline_size` and then remove the `flatten` call?

Reply:
The call two lines above is `LengthPercentageOrAuto::map`, not `Option::map`. Maybe we should have `LengthPercentageOrAuto::and_then` that return `Auto` when its closure returns `Option::None`?